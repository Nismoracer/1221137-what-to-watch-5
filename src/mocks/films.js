import {MovieRate} from "../const";

export default [
  {
    id: `1`,
    icon: `img/no-country-for-old-men.jpg`,
    background: `img/bg-the-grand-budapest-hotel.jpg`,
    poster: `img/the-grand-budapest-hotel-poster.jpg`,
    title: `Bohemian Rhapsody`,
    src: `https://download.blender.org/durian/trailer/sintel_trailer-480p.mp4`,
    mark: 8.2,
    rate: MovieRate.GOOD,
    ratings: `152`,
    director: `Quentin Tarantino`,
    actors: [`Sergey Bodrov`, `Aleksandr Bezrukov`, `Konstantin Khabensky`],
    details: {
      duration: `144`,
      genre: `comedy`,
      release: `2019`,
      comments: [`1`, `2`, `3`, `4`],
      description: `Norm film!`
    },
    isMyList: true
  },
  {
    id: `2`,
    icon: `img/aviator.jpg`,
    background: `img/bg-the-grand-budapest-hotel.jpg`,
    poster: `img/the-grand-budapest-hotel-poster.jpg`,
    title: `Pulp Fiction`,
    src: `https://download.blender.org/durian/trailer/sintel_trailer-480p.mp4`,
    mark: 9.1,
    rate: MovieRate.VERY_GOOD,
    ratings: `316`,
    director: `Quentin Tarantino`,
    actors: [`John Travolta`, `Bruce Willis`, `Uma Turman`],
    details: {
      duration: `120`,
      genre: `action`,
      release: `1998`,
      comments: [`5`, `6`],
      description: `Gangsta - to be, or not to be?`
    },
    isMyList: true
  },
  {
    id: `3`,
    icon: `img/bohemian-rhapsody.jpg`,
    background: `img/bg-the-grand-budapest-hotel.jpg`,
    poster: `img/the-grand-budapest-hotel-poster.jpg`,
    title: `Aviator`,
    src: `https://download.blender.org/durian/trailer/sintel_trailer-480p.mp4`,
    mark: 7.2,
    rate: MovieRate.NORMAL,
    ratings: `183`,
    director: `Martin Scorceso`,
    actors: [`Leonardo Di'Caprio`, `Jason Statham`],
    details: {
      duration: `80`,
      genre: `history`,
      release: `2002`,
      comments: [`7`, `8`, `9`, `10`],
      description: `Do you afraid to fly?`
    },
    isMyList: false
  },
  {
    id: `4`,
    icon: `img/midnight-special.jpg`,
    background: `img/bg-the-grand-budapest-hotel.jpg`,
    poster: `img/the-grand-budapest-hotel-poster.jpg`,
    title: `Snatch`,
    src: `https://download.blender.org/durian/trailer/sintel_trailer-480p.mp4`,
    mark: 8.6,
    rate: MovieRate.VERY_GOOD,
    ratings: `183`,
    director: `Martin Scorceso`,
    actors: [`Brett Pitt`, `Jason Statham`],
    details: {
      duration: `120`,
      genre: `action`,
      release: `2001`,
      comments: [`11`],
      description: `Do you like doggy?`
    },
    isMyList: false
  },
  {
    id: `5`,
    icon: `img/orlando.jpg`,
    background: `img/bg-the-grand-budapest-hotel.jpg`,
    poster: `img/the-grand-budapest-hotel-poster.jpg`,
    title: `No country for old man`,
    src: `https://download.blender.org/durian/trailer/sintel_trailer-480p.mp4`,
    mark: 8.1,
    rate: MovieRate.GOOD,
    ratings: `92`,
    director: `Alexandro Gonzales Inyarittu`,
    actors: [`Khavier Bardem`, `Nicholas Cage`],
    details: {
      duration: `102`,
      genre: `drama`,
      release: `2003`,
      comments: [`12`, `13`],
      description: `About a man and his killer`
    },
    isMyList: true
  },
  {
    id: `6`,
    icon: `img/pulp-fiction.jpg`,
    background: `img/bg-the-grand-budapest-hotel.jpg`,
    poster: `img/the-grand-budapest-hotel-poster.jpg`,
    title: `Zhmurki`,
    src: `https://download.blender.org/durian/trailer/sintel_trailer-480p.mp4`,
    mark: 7.2,
    rate: MovieRate.NORMAL,
    ratings: `125`,
    director: `Nikita Mikhalkov`,
    actors: [`Nikita Mikhalkov`, `Andrey Panin`],
    details: {
      duration: `124`,
      genre: `comedy`,
      release: `2008`,
      comments: [`14`, `15`, `16`],
      description: `Russia. 90-s.`
    },
    isMyList: false
  },
  {
    id: `7`,
    icon: `img/seven-years-in-tibet.jpg`,
    background: `img/bg-the-grand-budapest-hotel.jpg`,
    poster: `img/the-grand-budapest-hotel-poster.jpg`,
    title: `Fear and Loathing in Los Vegas`,
    src: `https://download.blender.org/durian/trailer/sintel_trailer-480p.mp4`,
    mark: 7.4,
    rate: MovieRate.GOOD,
    ratings: `145`,
    director: `Luc Besson`,
    actors: [`Jonny Depp`],
    details: {
      duration: `114`,
      genre: `comedy`,
      release: `1996`,
      comments: [`17`],
      description: `Desert, drugs and rock'n'roll`
    },
    isMyList: false
  },
  {
    id: `8`,
    icon: `img/what-we-do-in-the-shadows.jpg`,
    background: `img/bg-the-grand-budapest-hotel.jpg`,
    poster: `img/the-grand-budapest-hotel-poster.jpg`,
    title: `La-la-land`,
    src: `https://download.blender.org/durian/trailer/sintel_trailer-480p.mp4`,
    mark: 5.7,
    rate: MovieRate.BAD,
    ratings: `95`,
    director: `Olga Buzova`,
    actors: [`Olga Buzova`],
    details: {
      duration: `60`,
      genre: `musicle`,
      release: `2019`,
      comments: [`18`, `19`],
      description: `Desert, drugs and rock'n'roll`
    },
    isMyList: false
  },
];
