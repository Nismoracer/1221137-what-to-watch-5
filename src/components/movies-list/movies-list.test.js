import React from "react";
import renderer from "react-test-renderer";
import MoviesList from "./movies-list";
import films from "../../mocks/films";

it(`Should render Movies list correctly`, () => {
  const tree = renderer
    .create(
        <MoviesList
          films={films}
          onMovieClick={()=>{}}
        />
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});
