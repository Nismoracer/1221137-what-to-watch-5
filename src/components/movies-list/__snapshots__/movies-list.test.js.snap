// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should render Movies list correctly 1`] = `
<div
  className="catalog__movies-list"
>
  <article
    className="small-movie-card catalog__movies-card"
  >
    <div
      className="small-movie-card__image"
      onClick={[Function]}
      onMouseEnter={[Function]}
      onMouseLeave={[Function]}
    >
      <video
        height="100%"
        muted={true}
        width="100%"
      />
    </div>
    <h3
      className="small-movie-card__title"
    >
      <a
        className="small-movie-card__link"
        href="movie-page.html"
      >
        Bronson
      </a>
    </h3>
  </article>
  <article
    className="small-movie-card catalog__movies-card"
  >
    <div
      className="small-movie-card__image"
      onClick={[Function]}
      onMouseEnter={[Function]}
      onMouseLeave={[Function]}
    >
      <video
        height="100%"
        muted={true}
        width="100%"
      />
    </div>
    <h3
      className="small-movie-card__title"
    >
      <a
        className="small-movie-card__link"
        href="movie-page.html"
      >
        Pulp Fiction
      </a>
    </h3>
  </article>
  <article
    className="small-movie-card catalog__movies-card"
  >
    <div
      className="small-movie-card__image"
      onClick={[Function]}
      onMouseEnter={[Function]}
      onMouseLeave={[Function]}
    >
      <video
        height="100%"
        muted={true}
        width="100%"
      />
    </div>
    <h3
      className="small-movie-card__title"
    >
      <a
        className="small-movie-card__link"
        href="movie-page.html"
      >
        Midnight Special
      </a>
    </h3>
  </article>
  <article
    className="small-movie-card catalog__movies-card"
  >
    <div
      className="small-movie-card__image"
      onClick={[Function]}
      onMouseEnter={[Function]}
      onMouseLeave={[Function]}
    >
      <video
        height="100%"
        muted={true}
        width="100%"
      />
    </div>
    <h3
      className="small-movie-card__title"
    >
      <a
        className="small-movie-card__link"
        href="movie-page.html"
      >
        Matrix
      </a>
    </h3>
  </article>
  <article
    className="small-movie-card catalog__movies-card"
  >
    <div
      className="small-movie-card__image"
      onClick={[Function]}
      onMouseEnter={[Function]}
      onMouseLeave={[Function]}
    >
      <video
        height="100%"
        muted={true}
        width="100%"
      />
    </div>
    <h3
      className="small-movie-card__title"
    >
      <a
        className="small-movie-card__link"
        href="movie-page.html"
      >
        Legend
      </a>
    </h3>
  </article>
  <article
    className="small-movie-card catalog__movies-card"
  >
    <div
      className="small-movie-card__image"
      onClick={[Function]}
      onMouseEnter={[Function]}
      onMouseLeave={[Function]}
    >
      <video
        height="100%"
        muted={true}
        width="100%"
      />
    </div>
    <h3
      className="small-movie-card__title"
    >
      <a
        className="small-movie-card__link"
        href="movie-page.html"
      >
        Gangs of new york
      </a>
    </h3>
  </article>
  <article
    className="small-movie-card catalog__movies-card"
  >
    <div
      className="small-movie-card__image"
      onClick={[Function]}
      onMouseEnter={[Function]}
      onMouseLeave={[Function]}
    >
      <video
        height="100%"
        muted={true}
        width="100%"
      />
    </div>
    <h3
      className="small-movie-card__title"
    >
      <a
        className="small-movie-card__link"
        href="movie-page.html"
      >
        A Star Is Born
      </a>
    </h3>
  </article>
  <article
    className="small-movie-card catalog__movies-card"
  >
    <div
      className="small-movie-card__image"
      onClick={[Function]}
      onMouseEnter={[Function]}
      onMouseLeave={[Function]}
    >
      <video
        height="100%"
        muted={true}
        width="100%"
      />
    </div>
    <h3
      className="small-movie-card__title"
    >
      <a
        className="small-movie-card__link"
        href="movie-page.html"
      >
        Beach
      </a>
    </h3>
  </article>
  <article
    className="small-movie-card catalog__movies-card"
  >
    <div
      className="small-movie-card__image"
      onClick={[Function]}
      onMouseEnter={[Function]}
      onMouseLeave={[Function]}
    >
      <video
        height="100%"
        muted={true}
        width="100%"
      />
    </div>
    <h3
      className="small-movie-card__title"
    >
      <a
        className="small-movie-card__link"
        href="movie-page.html"
      >
        Aviator
      </a>
    </h3>
  </article>
  <article
    className="small-movie-card catalog__movies-card"
  >
    <div
      className="small-movie-card__image"
      onClick={[Function]}
      onMouseEnter={[Function]}
      onMouseLeave={[Function]}
    >
      <video
        height="100%"
        muted={true}
        width="100%"
      />
    </div>
    <h3
      className="small-movie-card__title"
    >
      <a
        className="small-movie-card__link"
        href="movie-page.html"
      >
        Johnny English
      </a>
    </h3>
  </article>
  <article
    className="small-movie-card catalog__movies-card"
  >
    <div
      className="small-movie-card__image"
      onClick={[Function]}
      onMouseEnter={[Function]}
      onMouseLeave={[Function]}
    >
      <video
        height="100%"
        muted={true}
        width="100%"
      />
    </div>
    <h3
      className="small-movie-card__title"
    >
      <a
        className="small-movie-card__link"
        href="movie-page.html"
      >
        Dardjeeling Limited
      </a>
    </h3>
  </article>
  <article
    className="small-movie-card catalog__movies-card"
  >
    <div
      className="small-movie-card__image"
      onClick={[Function]}
      onMouseEnter={[Function]}
      onMouseLeave={[Function]}
    >
      <video
        height="100%"
        muted={true}
        width="100%"
      />
    </div>
    <h3
      className="small-movie-card__title"
    >
      <a
        className="small-movie-card__link"
        href="movie-page.html"
      >
        Macbeth
      </a>
    </h3>
  </article>
  <article
    className="small-movie-card catalog__movies-card"
  >
    <div
      className="small-movie-card__image"
      onClick={[Function]}
      onMouseEnter={[Function]}
      onMouseLeave={[Function]}
    >
      <video
        height="100%"
        muted={true}
        width="100%"
      />
    </div>
    <h3
      className="small-movie-card__title"
    >
      <a
        className="small-movie-card__link"
        href="movie-page.html"
      >
        Snatch
      </a>
    </h3>
  </article>
  <article
    className="small-movie-card catalog__movies-card"
  >
    <div
      className="small-movie-card__image"
      onClick={[Function]}
      onMouseEnter={[Function]}
      onMouseLeave={[Function]}
    >
      <video
        height="100%"
        muted={true}
        width="100%"
      />
    </div>
    <h3
      className="small-movie-card__title"
    >
      <a
        className="small-movie-card__link"
        href="movie-page.html"
      >
        No Country for Old Men
      </a>
    </h3>
  </article>
  <article
    className="small-movie-card catalog__movies-card"
  >
    <div
      className="small-movie-card__image"
      onClick={[Function]}
      onMouseEnter={[Function]}
      onMouseLeave={[Function]}
    >
      <video
        height="100%"
        muted={true}
        width="100%"
      />
    </div>
    <h3
      className="small-movie-card__title"
    >
      <a
        className="small-movie-card__link"
        href="movie-page.html"
      >
        The Revenant
      </a>
    </h3>
  </article>
</div>
`;
